src/IO/S3/PocoHTTPClient.cpp

```diff
@@ -106,6 +106,17 @@ PocoHTTPClientConfiguration::PocoHTTPClientConfiguration(
 {
 }
 
+std::shared_ptr<PocoHTTPClient> PocoHTTPClientConfiguration::createAndGetClient() const
+{
+    http_client = std::make_shared<PocoHTTPClient>(*this);
+    return http_client;
+}
+
+std::shared_ptr<PocoHTTPClient> PocoHTTPClientConfiguration::getClient() const
+{
+    return http_client;
+}
```

Comment:
This is all about to make reachable PocoHTTPClient from DB::Client.
And make that right call
`http_client->RetryRequestSleep(std::chrono::milliseconds(sleepMillis));`

But it is quite controversial approach. Also the `doRequestWithRetryNetworkErrors` is questionable. It is better to have a good library with good retries that have fun with such patches.

So I'm going to not do sleeps at all inside `doRequestWithRetryNetworkErrors` and keep things simple.
`doRequestWithRetryNetworkErrors` is for quite specific and rare case and it is not supposed to be retried often.